syntax = "proto3";

message CardPool {
    string id = 1;
    repeated Card cards = 2;
    int32 deck_size = 3;
}

message Deck {
    string pool_id = 1;
    repeated int32 card_ids = 2;
}

message DeckResult {
    float win_rate = 1;
    int32 wins = 2;
    int32 losses = 3;
    int32 rank = 4;
    float percentile = 5;
    int32 simple_iron = 6;
    int32 simple_crystal = 7;
    int32 simple_iron_winner = 8;
    int32 simple_crystal_winner = 9;
}

// This card is for describing all displayable information to the client so that they can show the card to the user.
// Card functionality is all done server side right now so that is not transfered at all.
message Card {
    reserved 2;

    int32 id = 1;
    int32 iron_strength = 3;
    int32 crystal_strength = 4;

    enum Age {
        reserved 1;

        NONE_AGE = 0;
        // STONE = 1;
        IRON = 2;
        CRYSTAL = 3;
    }
    Age age = 5;

    enum Race {
        NONE_RACE = 0;
        BEASTMAN = 1;
        HUMAN = 2;
        UNDEAD = 3;
    }
    Race race = 6;

    enum Profession {
        NONE_PROF = 0;
        ALCHEMIST = 1;
        BATTLETECH = 2;
        CONJUROR = 3;
        PROPHET = 4;
        WOODSMAN = 5;
        PEASANT = 6;
    }
    Profession prof = 7;

    string desc = 8;
    Race race_synergy = 9;
    Profession prof_synergy = 10;
}